### DISCUSSION TITLE
Truffle Data Seeding

### DISCUSSION THREAD

**Discussion by ra-sun-god** [ConsenSys-archive/truffle#3988] [General Q&A]
Truffle is an awesome toolkit for every solidity and dapp developer, we at LibertyPie make advanced use of the truffle framework.  
During our core product development, we needed a simple way to seed initial data to different contracts similar to database seeding. <br />
Truffle has great migration support but no seeding of data out of the box.  
So we decided to build a simple CLI tool as an npm package for seeding initial data out of the box with 0 configurations.   <br>

Introducing [Liberty Pie's Truffle Seeder](https://github.com/LibertyPie/truffle-seeder), your easy way to seed initial data to any smart contract. Kindly help us improve it by giving us your feedback

Project: [https://github.com/LibertyPie/truffle-seeder](https://github.com/LibertyPie/truffle-seeder)

**Comment by cds-amal** [comment]
Hi @big-zak ,

Apologies for not responding sooner. We've been using `issues` as the primary mechanism to introduce ideas, but perhaps we should utilize this as well.

`truffle-seeder` looks like a useful addition to testing. Thank you for sharing! We'd love to promote its usage and I believe @kevinbluer has reached out already. 

I'm curious about where in the testing you inject the tests. is it scripted with an unmanaged ganache instance, or does it utilize a before test hook?
